src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py

```diff
@@ -1,14 +1,172 @@
+from __future__ import annotations
+
 from pathlib import Path
-from warnings import warn
+from typing import Literal, Optional
 
 import numpy as np
 import scipy
+from pynwb import NWBFile
 
+from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 from ..basesortingextractorinterface import BaseSortingExtractorInterface
 from ....tools import get_package
 from ....utils import FilePathType
 
 
+class CellExplorerRecordingInterface(BaseRecordingExtractorInterface):
+    """
+    This interface serves as an temporary solution for integrating CellExplorer metadata during a conversion process.
+
+    CellExplorer's new format (https://cellexplorer.org/) contains a `session.mat` file, which has the following field:
+
+    * Sampling frequency
+    * Gains for both raw data (held in a file named session.dat) and lfp (located in session.lfp)
+    * Dtype for both raw data and lfp.
+
+    Link to the documentation detailing the file's structure:
+    https://cellexplorer.org/datastructure/data-structure-and-format/#session-metadata
+
+    If the binary file is available (session.dat), the interface will use the `BinaryRecordingExtractor` to load the
+    data. Otherwise, it will use the `NumpyRecording` extractor, which is a dummy extractor that only contains
+    metadata. This can be used to add the electrode using `write_electrical_series=False` as a conversion option.
+
+    In its current form, the `chanMap.mat` file is employed to extract the electrode coordinates within the probe.
+    To my understanding, this file is generated for using kilosort, and it is not granted that it will
+    be universally available for all future conversion projects. Ideally, the format should also contain
+    files for channel info and channel coordinates, although these are absent in the current conversion:
+
+    https://cellexplorer.org/datastructure/data-structure-and-format/#channels
+
+    As we accumulate more examples moving forward, we will be able to figure out what's better. This is a first
+    iteration.
+    """
+
+    sampling_frequency_key = "sr"
+
+    def __init__(self, folder_path, verbose=True, es_key: str = "ElectricalSeries"):
```

Comment:
Though I think I would actually prefer two separate file-path type arguments; one for the binary and one for the header. Folder mode is overkill for that I'd say

Reply:
There is no such a clean distinction here. The relevant metadata is in a couple of files:

The metadata is in:
* `channelinfo.mat`
* `chanCoords`
* `chanMap` (if they used kilosort like in the current conversion)

The binary files are either:

* `{session}.dat`
* `session}.lfp`

Reply:
Oh I see, there's effectively more than one header file...

OK then